components/feature/recentlyclosed/src/main/java/mozilla/components/feature/recentlyclosed/ext/SessionManager.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.recentlyclosed.ext
+
+import android.content.Context
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.recentlyclosed.RecentlyClosedTab
+
+/**
+ * Restores the given [RecentlyClosedTab]. Will invoke [onTabRestored] on successful restore
+ * and [onFailure] otherwise.
+ */
+fun SessionManager.restoreRecentlyClosedTab(
```

Comment:
Let's move this functionality to work with the `BrowserStore` if possible since we want to move to that and the `SessionManager` is almost defunct.